src/komet/kdist/soroban-semantics/cheatcodes.md

```diff
@@ -34,6 +34,37 @@ module CHEATCODES
       requires getTag(HostVal(NEW_SEQ_NUM)) ==Int getTag(U32(0)) // check `NEW_SEQ_NUM` is a U32
 ```
 
+## kasmer_set_ledger_timestamp
+
+```rust
+    extern "C" {
+        fn kasmer_set_ledger_timestamp(x : u64);
+    }
+```
+
+```k
+    rule [kasmer-set-ledger-timestamp]:
+        <instrs> hostCall ( "env" , "kasmer_set_ledger_timestamp" , [ i64  .ValTypes ] -> [ .ValTypes ] )
+              => loadObject(HostVal(TIMESTAMP))
+              ~> setLedgerTimestamp
+              ~> toSmall(Void)
```

Comment:
@bbyalcinkaya, could you please elaborate a bit on the presence of `toSmall` at the end of the rewriting operation? After going through its implementation, it is not immediately clear to me why we should leave it at the K cell at the end of certain operations.

Reply:
My current understanding is that it is working as some sort of a return value for this host function (as in it is returning nothing, in this case), but I'd like to double check.

Reply:
@ACassimiro That's correct, it is used as a return value here. `toSmall` converts `ScVal`s to `HostVal`s. When a `HostVal` is on top of the `<instrs>` cell, it is executed as `i64.const _` instruction: [rule](https://github.com/runtimeverification/komet/blob/fxdao-hostcalls/src/komet/kdist/soroban-semantics/soroban.md?plain=1#L162).